src/ldp.rs

```diff
@@ -159,6 +176,12 @@ impl LinkedDataProofs {
                 x509_thumbprint_sha256: _,
             } if ec_params.curve == Some("secp256k1".to_string()) => {
                 if algorithm.as_ref() == Some(&Algorithm::ES256KR) {
+                    #[cfg(feature = "keccak-hash")]
+                    if let Some(ref vm) = options.verification_method {
+                        if vm.ends_with("#Eip712Method2021") {
```

Comment:
What's the best way to document our VM support? Are we okay with treating it as EcdsaSecp256k1RecoverySignature2020 if we don't compile in keccak support?

Reply:
Could this cause problems in some circumstances? I'm thinking that perhaps the feature flag should specify `eip712` and require keccak-hash, but happy to merge these changes and discuss this in an issue.

Reply:
This should probably instead fail here, since passing a VM ending in `#Eip712Method2021` is currently the only way for the caller to choose `Eip712Signature2021`. Ideally the proof type should be chosen by the `verificationMethod` option entirely instead of by the key type as this function currently does. During signing, we would deference the `verificationMethod` option to a verification method object (map) in the DID document, and then select the corresponding proof type, or error if it is unsupported.

Reply:
+1 to failing when in doubt